C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -53,8 +54,8 @@ public SlackAdapter(ISlackAdapterOptions options) : base()
 
             if (this.options.BotToken != null)
             {
-                Slack = new SlackAPI(this.options.BotToken);
-                Identity = Slack.GetIdentity();
+                Slack = new SlackTaskClient(this.options.BotToken);
+                Identity = Slack.MySelf?.id;
```

Comment:
Is it mandatory to use `Myself` as nullable? If it is actually null, does that mean that the client was not able to connect using the used token? If so, I would avoid using `Myself?` and maybe throw an exception.

Reply:
That syntax doesn't mean it's nullable. It means that, if `Slack.MySelf` is not null, it will assign `Slack.MySelf.id` to 'Identity'. It's the equivalent to first checking `Slack.MySelf` for null

Reply:
Slack.MySelf is _nullable_
```
The syntax T? is shorthand for Nullable<T>. The two forms are interchangeable.
```
More info [here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/)

Reply:
The class **_AuthTestResult_** may not be necessary anymore. Since we are using the properties of SlackTaskClient class instead. What do you think?

Reply:
Slack.MySelf is [not nullable](https://github.com/Inumedia/SlackAPI/blob/master/SlackAPI/SlackTaskClient.cs#L18) 

The operator in question here is not *?*, but *?.*, the [null conditional operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-)